grakn-engine/src/main/java/ai/grakn/engine/factory/EngineGraknGraphFactory.java

```diff
@@ -50,19 +54,29 @@ public static EngineGraknGraphFactory create(Properties properties) {
     private EngineGraknGraphFactory(Properties properties) {
         this.properties = new Properties();
         this.properties.putAll(properties);
+        this.engineURI = properties.getProperty(GraknEngineConfig.SERVER_HOST_NAME) + ":" + properties.getProperty(GraknEngineConfig.SERVER_PORT_NUMBER);
+        this.systemKeyspace = new SystemKeyspace(this);
     }
 
     public synchronized void refreshConnections(){
         FactoryBuilder.refresh();
     }
 
     public GraknGraph getGraph(String keyspace, GraknTxType type){
-        return FactoryBuilder.getFactory(keyspace, Grakn.DEFAULT_URI, properties).open(type);
+        if(!keyspace.equals(SystemKeyspace.SYSTEM_GRAPH_NAME)) {
```

Comment:
Yeah, I get what you are saying. Right now it is neither a singleton nor a static class but because we instantiate it in the server constructor and pass the object around, its alright.